SEP-0012.md

```diff
@@ -0,0 +1,343 @@
+# SEP-0012 -- NDCube 2
```

Comment:
```suggestion
# SEP-0012 -- ndcube 2
```

Reply:
This was actually delbrate. This API is the API of the `NDCube` object. Sure it pulls in the associated classes, but things like collection and sequence are explicitly out of scope here.

Reply:
AH, good to know.  That should be made explicit in the first sentence of the introduction.